pcsx2/CDVD/CsoFileReader.cpp

```diff
@@ -256,14 +264,23 @@ int CsoFileReader::ReadChunk(void* dst, s64 chunkID)
 		// This might be less bytes than frameRawSize in case of padding on the last frame.
 		// This is because the index positions must be aligned.
 		const u32 readRawBytes = fread(m_readBuffer.get(), 1, frameRawSize, m_src);
+		bool success = false;
 
-		m_z_stream->next_in = m_readBuffer.get();
-		m_z_stream->avail_in = readRawBytes;
-		m_z_stream->next_out = static_cast<Bytef*>(dst);
-		m_z_stream->avail_out = m_frameSize;
+		if (m_uselz4)
+		{
+			int res = LZ4_decompress_safe((const char*)(m_readBuffer.get()), (char*)dst, (int)readRawBytes, (int)m_frameSize);
+			success = res == (int)m_frameSize;
```

Comment:
Didn't see these, can you replace the C casts with static/reinterpret's please :)

Reply:
Fixed. Should be good now.